### ISSUE TITLE
Package request: darawich

### ISSUE THREAD

**Issue by Kernald** [NixOS/nixpkgs#345622]
**Project description**

Darawich is a self-hosted alternative to Google Location History/Maps timeline, supporting multiple data-sources (including OwnTracks, Immich, Google Maps...)

**Metadata**

* homepage URL: https://dawarich.app/
* source URL: https://github.com/Freika/dawarich
* license: AGPL 3.0
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by mainrs** [comment]
Does somebody know of projects already packaged into nixpkgs using the same stack that I can use as a template?

**Comment by diogotcorreia** [comment]
@mainrs I don't know any, but I was waiting on upstream to publish a production configuration before trying to package this
https://github.com/Freika/dawarich/pull/437

We can probably already rely on the changes in the PR tho.

**Comment by diogotcorreia** [comment]
> Does somebody know of projects already packaged into nixpkgs using the same stack that I can use as a template?

Started taking a look into this, Mastodon seems to be quite a similar stack. https://github.com/NixOS/nixpkgs/tree/master/pkgs/servers/mastodon

It appears that upstream's `Gemfile.lock` does not yet have checksums (and there is not support for that yet in nixpkgs, see #372597), so we will need a `gemset.nix` file just like Mastodon.

I have never used Ruby before so I'm very unfamiliar with the ecosystem, it would be great if someone could package this instead. Otherwise, if no one steps up, I can perhaps give it a try in a few weeks. I'll be happy to test any attempts to package this.

**Comment by mainrs** [comment]
@diogotcorreia I'll try to get this working similar to how mastadon does it. I'll ping you here in a new comment to let you know!

**Comment by diogotcorreia** [comment]
@mainrs Have you made any progress on this? Anything we can help with?

**Comment by mainrs** [comment]
@diogotcorreia I had to postpone the implementation due to my thesis D: I did modify the mastadon approach but hit some issues and stopped working on it. I tried to find the old repository on my machine, but I think I deleted the repository.

**Comment by diogotcorreia** [comment]
@mainrs I see. I'll take my turn to procrastinate my thesis and I'll give it a try :) Will open a draft PR as soon as I have something meaningful.

**Comment by diogotcorreia** [comment]
I've made some progress: https://github.com/NixOS/nixpkgs/pull/423867

It seems to turn on and I can login. Haven't tried much else, but at least seems to be working somewhat. There's still a lot to polish tho, but it's a good start.
